app/src/main/java/pl/temomuko/autostoprace/ui/settings/SettingsPresenter.java

```diff
@@ -35,25 +38,25 @@ public void detachView() {
     }
 
     public void setupLogoutPreference() {
-        boolean isAuth = mDataManager.isLoggedWithToken();
+        boolean isAuth = dataManager.isLoggedWithToken();
         getMvpView().setupLogoutPreferenceEnabled(isAuth);
         if (isAuth) {
-            String username = mDataManager.getCurrentUser().getUsername();
+            String username = dataManager.getCurrentUser().getUsername();
             getMvpView().setupUserLogoutPreferenceSummary(username);
         } else {
             getMvpView().setupGuestLogoutPreferenceSummary();
         }
     }
 
     public void logout() {
-        mSubscription = mDataManager.signOut()
-                .compose(RxUtil.applyIoSchedulers())
-                .subscribe(response -> {
-                    LogUtil.i(TAG, response.body().toString());
+        mSubscription = authenticator.logout()
+                .compose(RxUtil.applyCompletableIoSchedulers())
+                .doOnTerminate(() -> dataManager.clearUserData().subscribe())
```

Comment:
isnt there some do after bla bla operator in rx1 instead of doing some inner subscription?